File: UpdatedPractice2025/Arrays_And_Hashing/contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 * Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.
 */
const containsDuplicate = function(nums) {
    const visitedNums = new Map();

    for (const num of nums) {
        const numVisited = visitedNums.get(num)
        if (numVisited) {
            return true
        }
        visitedNums.set(num, true)
    }

    return false
};

// Time complexity: O(n)
// Space complexity: O(n)

File: UpdatedPractice2025/Arrays_And_Hashing/contains-duplicate.go
// Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

func containsDuplicate(nums []int) bool {
    numsVisited := make(map[int]bool)

    for _, num := range nums { 
        if _, ok := numsVisited[num]; ok {
            return true
        }
        numsVisited[num] = true
    }
    
    return false;
}

// Time complexity: O(n)
// Space complexity: O(n)